### ISSUE TITLE
Using EM-WebSocket w/ AMQP

### ISSUE THREAD

**Issue by clintmiller** [igrigorik/em-websocket#25]
I'm having some problems using AMQP (message queuing) with EM-WebSocket.

Here's the basic idea:

```
EventMachine.run {
  EventMachine::WebSocket.start(:host => "0.0.0.0", :port => 8080) do |socket|
    socket.onopen {
      puts "WS connection opened"
      AMQP.start(:host => "hostname") do
        exchange = MQ.topic("exchange.name", :durable => true)
        mq = MQ.new 
        mq.queue("queue.name").bind(exchange).subscribe do |m|
          puts "handling new message: #{m}"
          socket.send m
        end
      end
    }
  }
}
```

This works well enough, in that I see the browser (Chrome) connect to the socket server and that the socket server is able to consume messages from the AMQP queue. I run into problems though when multiple messages are waiting in the queue for pickup. Sometimes, it seems like all the messages are picked up, framed correctly, and sent through the socket to the browser. Other times, it seems like all the messages are picked up, but some number (_1..n-1_, _n_ being the number of messages available) are framed correctly and sent through the socket to the browser.

In short, it seems like all the messages are correctly consumed from the queue, but not all are consistently sent to the browser (though the ones that are sent are framed correctly).

I have confirmed the actual data transmitted through the socket with a proxy, so I'm fairly confident this describes the real problem.
